src/test/gather_node_test.cpp

```diff
@@ -0,0 +1,189 @@
+        },
+        {
+            "config": {
+                "name": "dummy2bs",
+                "base_path": "/ovms/src/test/dummy",
+                "target_device": "CPU",
+                "model_version_policy": {"all": {}},
+                "nireq": 1,
+                "batch_size": 2
+            }
+        }
+    ]
+})";
+
+class DLNodeSessionWithGetInputsExposed : public DLNodeSession {
+public:
+    DLNodeSessionWithGetInputsExposed(const NodeSessionMetadata& metadata, const std::string& nodeName, uint32_t inputsCount, session_id_t shardsCount, ModelManager& manager, const std::string& modelName, model_version_t modelVersion) :
+        DLNodeSession(metadata, nodeName, inputsCount, shardsCount, manager, modelName, modelVersion) {}
+    DLNodeSessionWithGetInputsExposed(const NodeSessionMetadata&& metadata, const std::string& nodeName, uint32_t inputsCount, session_id_t shardsCount, ModelManager& manager, const std::string& modelName, model_version_t modelVersion) :
+        DLNodeSession(std::move(metadata), nodeName, inputsCount, shardsCount, manager, modelName, modelVersion) {}
+
+    const auto& getInputs() const {
+        return this->inputHandler->getInputs();
+    }
+};
+
+class DLNodeWithGetInputsExposed : public DLNode {
+public:
+    DLNodeWithGetInputsExposed(const std::string& nodeName, const std::string& modelName, std::optional<model_version_t> modelVersion,
+        ModelManager& modelManager,
+        std::unordered_map<std::string, std::string> nodeOutputNameAlias, const std::optional<std::set<std::string>>& gatherFrom) :
+        DLNode(nodeName, modelName, modelVersion, modelManager, nodeOutputNameAlias) {
+        this->gatherFrom = gatherFrom;
+    }
+    const auto& getInputsFromInputHandler(session_key_t sessionId) const {
+        DLNodeSessionWithGetInputsExposed& dlnodesessionWithGetInputsExposed = static_cast<DLNodeSessionWithGetInputsExposed&>(*nodeSessions.at(sessionId));
+        return dlnodesessionWithGetInputsExposed.getInputs();
+    }
+    std::unique_ptr<NodeSession> createNodeSession(const NodeSessionMetadata& metadata, session_id_t shardsCount) override {
+        return std::make_unique<DLNodeSessionWithGetInputsExposed>(metadata, getName(), previous.size(), shardsCount,
+            this->modelManager, this->modelName, this->modelVersion.value_or(0));
+    }
+};
+
+TEST_F(GatherNodeTest, FullFlowGatherInNonExitNode) {
+    // This test simulates node with multiple subsessions connected to following node
+    // that should gather it results but is not exit node
+    ConstructorEnabledModelManager manager;
+    const std::string fileToReload = directoryPath + "/ovms_config_file.json";
+    createConfigFileWithContent(configDummy1BsDummy2Bs, fileToReload);
+    ASSERT_EQ(manager.loadConfig(fileToReload), StatusCode::OK);
+    const std::string node1Name = "node1";
+    DLNode oneDummyNode1{node1Name, "dummy", 1, manager, {}};
+    const std::string demultiplexerNodeName{"nodeDummy"};
```

Comment:
There is no node with such name in pipeline, am I right?

Reply:
Yes - its existence is not required for this test to work. It bypasses the pipeline definition, pipeline creation & validation steps  .